async_simple/CMakeLists.txt

```diff
@@ -1,53 +1,65 @@
 file(GLOB coro_src "coro/*.cpp")
 file(GLOB executors_src "executors/*.cpp")
-file(GLOB uthread_src "uthread/internal/*.cc")
 
-EXECUTE_PROCESS(COMMAND uname -m COMMAND tr -d '\n' OUTPUT_VARIABLE ARCHITECTURE)
-if ("${ARCHITECTURE}" STREQUAL "aarch64")
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux") # uname -s
+  file(GLOB uthread_src "uthread/internal/*.cc")
+  if ("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "aarch64") # uname -p
     file(GLOB uthread_asm_src "uthread/internal/*arm64_aapcs_elf*")
-elseif ("${ARCHITECTURE}" STREQUAL "x86_64")
+    set(UTHREAD ON)
+  elseif ("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
     file(GLOB uthread_asm_src "uthread/internal/*x86_64_sysv_elf*")
-else()
-    message(FATAL_ERROR "Unsupported Target: ${ARCHITECTURE}")
+    set(UTHREAD ON)
+  else()
+    message(STATUS "Uthread Unsupported Target: ${CMAKE_SYSTEM_PROCESSOR}")
+  endif()
 endif()
 
 file(GLOB headers "*.h")
 file(GLOB coro_header "coro/*.h")
 file(GLOB executors_header "executors/*.h")
 file(GLOB experimental_header "experimental/*.h")
 file(GLOB util_header "util/*.h")
-file(GLOB uthread_header "uthread/*.h")
-file(GLOB uthread_internal_header "uthread/internal/*.h")
+if(UTHREAD)
+  file(GLOB uthread_header "uthread/*.h")
+  file(GLOB uthread_internal_header "uthread/internal/*.h")
+endif()
 
-message(STATUS "uthread_asm_src: ${uthread_asm_src}")
 
 set(SRCS
-    ${coro_src}
-    ${executors_src}
-    ${uthread_asm_src}
-    ${uthread_src}
-    )
+  ${coro_src}
+  ${executors_src}
+  )
+if(UTHREAD)
+  list(APPEND SRCS ${uthread_src})
+  list(APPEND SRCS ${uthread_asm_src})
+endif()
 
-add_library(async_simple_static STATIC ${SRCS})
-add_library(async_simple  SHARED ${SRCS})
-target_link_libraries(async_simple ${deplibs})
-target_link_libraries(async_simple_static ${deplibs})
+if(SRCS)
```

Comment:
Is this condition necessary?

Reply:
the SRCS is null after disabled uthread
cmake will throw error if SRCS is empty

Reply:
How about adding an empty src file to reduce the branches? I feel things is more uniform in this way.

Reply:
i prefer to move some not effect performance logic to cpp source, empty src file so ugly

Reply:
how about to rename LazyHelper.h to syncAwait.cpp, and move syncAwait declare to Lazy.h

Reply:
Event.h not template class, we can move some implement to cpp

Reply:
CountEvent contains some atomic operations. I feel it might affect the performance. Oh, if you feel empty src file is ugly, I think we could reduce the branch later once we have actual src files.

Reply:
> CountEvent contains some atomic operations. I feel it might affect the performance. Oh, if you feel empty src file is ugly, I think we could reduce the branch later once we have actual src files.

OK，I will add a comment to make others note it